### PR TITLE
frontend: speedup the removal of temporary projects

### LINKED ISSUE #2489: Temporary projects with automated builds not deleted
I have a couple *temporary project: will be deleted ASAP* projects on Copr and they seem to be quite immortal.

https://copr.fedorainfracloud.org/coprs/churchyard/hypothesis-6.6/
https://copr.fedorainfracloud.org/coprs/churchyard/hypothesis-control/
https://copr.fedorainfracloud.org/coprs/churchyard/generators-extras-case/
https://copr.fedorainfracloud.org/coprs/churchyard/numpy-1.23/
https://copr.fedorainfracloud.org/coprs/churchyard/sequoia/
https://copr.fedorainfracloud.org/coprs/churchyard/clamp_mtime_to_source_date_epoch/
https://copr.fedorainfracloud.org/coprs/churchyard/flit-3.7.1/

Most of them have automated builds enabled, which might or might not be the reason they were not deleted.

### PR DESCRIPTION
**Pull request by FrostyX**
Fix #2489

I tried this on today's database dump and it took 46m24.731s to finish and my laptop was fine. I think the crash was prevented by not committing everything at once.

### CODE REVIEW

**Comment by FrostyX** [comment]
The F38 build fails because of

> ImportError: cannot import name 'batched' from 'itertools' (unknown location)

Should I work on the compatibility or can we ditch F38 for our server packages now?

File: frontend/coprs_frontend/commands/clean_expired_projects.py [hunk 1]
```diff
@@ -11,6 +11,5 @@ def clean_expired_projects():
     Clean all the expired temporary projects.  This command is meant to be
     executed by cron.
     """
-
-    with db_session_scope():
-        ComplexLogic.delete_expired_projects()
+    while ComplexLogic.delete_expired_projects(limit=100):
```
Comment (praiskup) [inline frontend/coprs_frontend/commands/clean_expired_projects.py:14]: do you want to use the limit here?

File: frontend/coprs_frontend/coprs/logic/complex_logic.py [hunk 2]
```diff
@@ -97,34 +98,43 @@ def delete_copr(cls, copr, admin_action=False):
...
             user = flask.g.user
 
-        builds_query = BuildsLogic.get_multiple_by_copr(copr=copr)
+        builds = [x.id for x in BuildsLogic.get_multiple_by_copr(copr=copr)]
 
         if copr.persistent:
             raise exceptions.InsufficientRightsException("This project is protected against deletion.")
 
-        for build in builds_query:
+        for chunk in batched(builds, 1000):
```
Comment (praiskup) [inline frontend/coprs_frontend/coprs/logic/complex_logic.py:106]: I'd suggest much smaller chunks ... consider projects with thousands of builds.

File: frontend/coprs_frontend/commands/clean_expired_projects.py (hunk 1 above)
Comment (praiskup) [reply frontend/coprs_frontend/commands/clean_expired_projects.py:14]: Nah, isn't this the reason why the script finished in 45 minutes?

**Comment by praiskup** [comment]
https://fedorapeople.org/groups/schedule/f-38/f-38-all-tasks.html EOL in May - let's drop the F38 support

File: frontend/coprs_frontend/coprs/logic/complex_logic.py [hunk 3]
```diff
@@ -97,34 +98,43 @@ def delete_copr(cls, copr, admin_action=False):
...
             # Don't send delete action for each build, rather send an action to delete
             # a whole project as a part of CoprsLogic.delete_unsafe() method.
-            BuildsLogic.delete_build(user, build, send_delete_action=False)
+            BuildsLogic.delete_builds(user, chunk, send_delete_action=False)
```
Comment (praiskup) [inline frontend/coprs_frontend/coprs/logic/complex_logic.py:109]: Where do we commit? At first glance, it seems we don't, and we still create a huge transaction.

File: frontend/coprs_frontend/commands/clean_expired_projects.py (hunk 1 above)
Comment (FrostyX) [reply frontend/coprs_frontend/commands/clean_expired_projects.py:14]: Yeah, I think we want the limit here. We can change the value though.

File: frontend/coprs_frontend/coprs/logic/complex_logic.py (hunk 3 above)
Comment (FrostyX) [reply frontend/coprs_frontend/coprs/logic/complex_logic.py:109]: We commit in the

File: frontend/coprs_frontend/commands/clean_expired_projects.py (hunk 1 above)
Comment (praiskup) [reply frontend/coprs_frontend/commands/clean_expired_projects.py:14]: ah, I missed the `s/with/while/` change, it seems OK then!  But don't we want to use a much smaller limit, like 5 :-)?

**Review by praiskup — APPROVED** [review]
LGTM

**Pull request merged**